From a6d1334da1867eaef650e865137caadca86c899f Mon Sep 17 00:00:00 2001
From: Camelia Groza <camelia.groza@nxp.com>
Date: Mon, 9 Dec 2019 18:20:05 +0200
Subject: [PATCH 707/741] sdk_dpaa: ceetm: coding style cleanup

commit 4599e639c96d7da2586ea64248ca9e574e87a71a from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Fix checkpatch warnings and use reverse christmas tree variable ordering
throughout the driver

Signed-off-by: Camelia Groza <camelia.groza@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 .../freescale/sdk_dpaa/dpaa_eth_ceetm.c       | 134 +++++++++---------
 1 file changed, 68 insertions(+), 66 deletions(-)

diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_ceetm.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_ceetm.c
index afd456b51150..8c8b18ad2cb0 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_ceetm.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_ceetm.c
@@ -48,10 +48,10 @@ struct Qdisc_ops ceetm_qdisc_ops;
 static void get_dcp_and_sp(struct net_device *dev, enum qm_dc_portal *dcp_id,
 			   unsigned int *sp_id)
 {
-	uint32_t channel;
-	t_LnxWrpFmPortDev *port_dev;
 	struct dpa_priv_s *dpa_priv = netdev_priv(dev);
 	struct mac_device *mac_dev = dpa_priv->mac_dev;
+	t_LnxWrpFmPortDev *port_dev;
+	uint32_t channel;
 
 	port_dev = (t_LnxWrpFmPortDev *)mac_dev->port_dev[TX];
 	channel = port_dev->txCh;
@@ -76,7 +76,7 @@ static void dpaa_drain_fqs(struct net_device *dev)
 	struct qman_fq *fq;
 	int ret, i;
 
-	for (i = 0; i < DPAA_ETH_TX_QUEUES; i ++) {
+	for (i = 0; i < DPAA_ETH_TX_QUEUES; i++) {
 		fq = priv->egress_fqs[i];
 		while (true) {
 			ret = qman_query_fq_np(fq, &np);
@@ -97,7 +97,7 @@ static void dpaa_drain_fqs(struct net_device *dev)
 static void ceetm_drain_class(struct ceetm_class *cl)
 {
 	struct qm_mcr_ceetm_cq_query cq_query;
-	struct qm_ceetm_cq *cq;
+	struct qm_ceetm_cq *cq = NULL;
 	unsigned int idx;
 	int ret;
 
@@ -109,10 +109,10 @@ static void ceetm_drain_class(struct ceetm_class *cl)
 		/* The ROOT classes aren't directly linked to CEETM CQs */
 		return;
 	case CEETM_PRIO:
-		cq = (struct qm_ceetm_cq*)cl->prio.cq;
+		cq = (struct qm_ceetm_cq *)cl->prio.cq;
 		break;
 	case CEETM_WBFS:
-		cq = (struct qm_ceetm_cq*)cl->wbfs.cq;
+		cq = (struct qm_ceetm_cq *)cl->wbfs.cq;
 		break;
 	}
 
@@ -191,10 +191,14 @@ static void ceetm_ern(struct qman_portal *portal, struct qman_fq *fq,
 /* Congestion State Change Notification callback */
 static void ceetm_cscn(struct qm_ceetm_ccg *ccg, void *cb_ctx, int congested)
 {
-	struct ceetm_fq *ceetm_fq = (struct ceetm_fq *)cb_ctx;
-	struct dpa_priv_s *dpa_priv = netdev_priv(ceetm_fq->net_dev);
-	struct ceetm_class *cls = ceetm_fq->ceetm_cls;
 	struct ceetm_class_stats *cstats = NULL;
+	struct dpa_priv_s *dpa_priv;
+	struct ceetm_fq *ceetm_fq;
+	struct ceetm_class *cls;
+
+	ceetm_fq = (struct ceetm_fq *)cb_ctx;
+	dpa_priv = netdev_priv(ceetm_fq->net_dev);
+	cls = ceetm_fq->ceetm_cls;
 
 	switch (cls->type) {
 	case CEETM_PRIO:
@@ -237,10 +241,10 @@ static int ceetm_config_ccg(struct qm_ceetm_ccg **ccg,
 			    struct qm_ceetm_channel *channel, unsigned int id,
 			    struct ceetm_fq *fq, struct dpa_priv_s *dpa_priv)
 {
-	int err;
-	u32 cs_th;
-	u16 ccg_mask;
 	struct qm_ceetm_ccg_params ccg_params;
+	u16 ccg_mask;
+	u32 cs_th;
+	int err;
 
 	err = qman_ceetm_ccg_claim(ccg, channel, id, ceetm_cscn, fq);
 	if (err)
@@ -285,9 +289,9 @@ static int ceetm_config_ccg(struct qm_ceetm_ccg **ccg,
 static int ceetm_config_lfq(struct qm_ceetm_cq *cq, struct ceetm_fq *fq,
 			    struct qm_ceetm_lfq **lfq)
 {
-	int err;
 	u64 context_a;
 	u32 context_b;
+	int err;
 
 	err = qman_ceetm_lfq_claim(lfq, cq);
 	if (err)
@@ -317,8 +321,8 @@ static int ceetm_config_prio_cls(struct ceetm_class *cls,
 				 struct net_device *dev,
 				 unsigned int id)
 {
-	int err;
 	struct dpa_priv_s *dpa_priv = netdev_priv(dev);
+	int err;
 
 	err = ceetm_alloc_fq(&cls->prio.fq, dev, cls);
 	if (err)
@@ -358,8 +362,8 @@ static int ceetm_config_wbfs_cls(struct ceetm_class *cls,
 				 struct net_device *dev,
 				 unsigned int id, int type)
 {
-	int err;
 	struct dpa_priv_s *dpa_priv = netdev_priv(dev);
+	int err;
 
 	err = ceetm_alloc_fq(&cls->wbfs.fq, dev, cls);
 	if (err)
@@ -518,11 +522,11 @@ static void ceetm_cls_destroy(struct Qdisc *sch, struct ceetm_class *cl)
 /* Destroy a ceetm qdisc */
 static void ceetm_destroy(struct Qdisc *sch)
 {
-	unsigned int ntx, i;
-	struct hlist_node *next;
-	struct ceetm_class *cl;
 	struct ceetm_qdisc *priv = qdisc_priv(sch);
 	struct net_device *dev = qdisc_dev(sch);
+	struct hlist_node *next;
+	struct ceetm_class *cl;
+	unsigned int ntx, i;
 
 	pr_debug(KBUILD_BASENAME " : %s : destroy qdisc %X\n",
 		 __func__, sch->handle);
@@ -593,13 +597,13 @@ static void ceetm_destroy(struct Qdisc *sch)
 
 static int ceetm_dump(struct Qdisc *sch, struct sk_buff *skb)
 {
+	struct ceetm_qdisc *priv = qdisc_priv(sch);
+	struct net_device *dev = qdisc_dev(sch);
+	struct ceetm_qdisc_stats *qstats;
+	struct tc_ceetm_qopt qopt;
 	struct Qdisc *qdisc;
 	unsigned int ntx, i;
 	struct nlattr *nest;
-	struct tc_ceetm_qopt qopt;
-	struct ceetm_qdisc_stats *qstats;
-	struct net_device *dev = qdisc_dev(sch);
-	struct ceetm_qdisc *priv = qdisc_priv(sch);
 
 	pr_debug(KBUILD_BASENAME " : %s : qdisc %X\n", __func__, sch->handle);
 
@@ -674,17 +678,20 @@ static int ceetm_init_root(struct Qdisc *sch, struct ceetm_qdisc *priv,
 			   struct tc_ceetm_qopt *qopt,
 			   struct netlink_ext_ack *extack)
 {
+	struct net_device *dev = qdisc_dev(sch);
+	unsigned int i, sp_id, parent_id;
 	struct netdev_queue *dev_queue;
-	struct Qdisc *qdisc;
+	struct dpa_priv_s *dpa_priv;
+	struct mac_device *mac_dev;
 	enum qm_dc_portal dcp_id;
-	unsigned int i, sp_id, parent_id;
+	struct qm_ceetm_lni *lni;
+	struct qm_ceetm_sp *sp;
+	struct Qdisc *qdisc;
 	int err;
 	u64 bps;
-	struct qm_ceetm_sp *sp;
-	struct qm_ceetm_lni *lni;
-	struct net_device *dev = qdisc_dev(sch);
-	struct dpa_priv_s *dpa_priv = netdev_priv(dev);
-	struct mac_device *mac_dev = dpa_priv->mac_dev;
+
+	dpa_priv = netdev_priv(dev);
+	mac_dev = dpa_priv->mac_dev;
 
 	pr_debug(KBUILD_BASENAME " : %s : qdisc %X\n", __func__, sch->handle);
 
@@ -712,9 +719,8 @@ static int ceetm_init_root(struct Qdisc *sch, struct ceetm_qdisc *priv,
 	priv->root.qdiscs = kcalloc(dev->num_tx_queues,
 				    sizeof(priv->root.qdiscs[0]),
 				    GFP_KERNEL);
-	if (!priv->root.qdiscs) {
+	if (!priv->root.qdiscs)
 		return -ENOMEM;
-	}
 
 	for (i = 0; i < dev->num_tx_queues; i++) {
 		dev_queue = netdev_get_tx_queue(dev, i);
@@ -812,11 +818,11 @@ static int ceetm_init_root(struct Qdisc *sch, struct ceetm_qdisc *priv,
 static int ceetm_init_prio(struct Qdisc *sch, struct ceetm_qdisc *priv,
 			   struct tc_ceetm_qopt *qopt)
 {
-	int err;
-	unsigned int i;
 	struct ceetm_class *parent_cl, *child_cl;
 	struct net_device *dev = qdisc_dev(sch);
 	struct Qdisc *root_qdisc = dev->qdisc;
+	unsigned int i;
+	int err;
 
 	pr_debug(KBUILD_BASENAME " : %s : qdisc %X\n", __func__, sch->handle);
 
@@ -853,11 +859,8 @@ static int ceetm_init_prio(struct Qdisc *sch, struct ceetm_qdisc *priv,
 	/* Create and configure qcount child classes */
 	for (i = 0; i < priv->prio.qcount; i++) {
 		child_cl = kzalloc(sizeof(*child_cl), GFP_KERNEL);
-		if (!child_cl) {
-			pr_err(KBUILD_BASENAME " : %s : kzalloc() failed\n",
-			       __func__);
+		if (!child_cl)
 			return -ENOMEM;
-		}
 
 		child_cl->prio.cstats = alloc_percpu(struct ceetm_class_stats);
 		if (!child_cl->prio.cstats) {
@@ -907,12 +910,12 @@ static int ceetm_init_prio(struct Qdisc *sch, struct ceetm_qdisc *priv,
 static int ceetm_init_wbfs(struct Qdisc *sch, struct ceetm_qdisc *priv,
 			   struct tc_ceetm_qopt *qopt)
 {
-	int err, group_b, small_group;
-	unsigned int i, id, prio_a, prio_b;
 	struct ceetm_class *parent_cl, *child_cl, *root_cl;
 	struct Qdisc *root_qdisc, *parent_qdisc = NULL;
-	struct ceetm_qdisc *root_priv;
 	struct net_device *dev = qdisc_dev(sch);
+	unsigned int i, id, prio_a, prio_b;
+	int err, group_b, small_group;
+	struct ceetm_qdisc *root_priv;
 
 	pr_debug(KBUILD_BASENAME " : %s : qdisc %X\n", __func__, sch->handle);
 
@@ -1070,11 +1073,8 @@ static int ceetm_init_wbfs(struct Qdisc *sch, struct ceetm_qdisc *priv,
 	/* Create qcount child classes */
 	for (i = 0; i < priv->wbfs.qcount; i++) {
 		child_cl = kzalloc(sizeof(*child_cl), GFP_KERNEL);
-		if (!child_cl) {
-			pr_err(KBUILD_BASENAME " : %s : kzalloc() failed\n",
-			       __func__);
+		if (!child_cl)
 			return -ENOMEM;
-		}
 
 		child_cl->wbfs.cstats = alloc_percpu(struct ceetm_class_stats);
 		if (!child_cl->wbfs.cstats) {
@@ -1138,11 +1138,11 @@ static int ceetm_init_wbfs(struct Qdisc *sch, struct ceetm_qdisc *priv,
 static int ceetm_init(struct Qdisc *sch, struct nlattr *opt,
 		      struct netlink_ext_ack *extack)
 {
-	struct tc_ceetm_qopt *qopt;
-	struct nlattr *tb[TCA_CEETM_QOPS + 1];
-	int ret;
 	struct ceetm_qdisc *priv = qdisc_priv(sch);
 	struct net_device *dev = qdisc_dev(sch);
+	struct nlattr *tb[TCA_CEETM_QOPS + 1];
+	struct tc_ceetm_qopt *qopt;
+	int ret;
 
 	pr_debug(KBUILD_BASENAME " : %s : qdisc %X\n", __func__, sch->handle);
 
@@ -1265,8 +1265,8 @@ static int ceetm_change_root(struct Qdisc *sch, struct ceetm_qdisc *priv,
 static int ceetm_change_wbfs(struct Qdisc *sch, struct ceetm_qdisc *priv,
 			     struct tc_ceetm_qopt *qopt)
 {
-	int err;
 	bool group_b;
+	int err;
 
 	if (qopt->qcount) {
 		pr_err("CEETM: the qcount can not be modified\n");
@@ -1324,11 +1324,11 @@ static int ceetm_change_wbfs(struct Qdisc *sch, struct ceetm_qdisc *priv,
 static int ceetm_change(struct Qdisc *sch, struct nlattr *opt,
 			struct netlink_ext_ack *extack)
 {
-	struct tc_ceetm_qopt *qopt;
-	struct nlattr *tb[TCA_CEETM_QOPS + 1];
-	int ret;
 	struct ceetm_qdisc *priv = qdisc_priv(sch);
 	struct net_device *dev = qdisc_dev(sch);
+	struct nlattr *tb[TCA_CEETM_QOPS + 1];
+	struct tc_ceetm_qopt *qopt;
+	int ret;
 
 	pr_debug(KBUILD_BASENAME " : %s : qdisc %X\n", __func__, sch->handle);
 
@@ -1381,8 +1381,8 @@ static int ceetm_change(struct Qdisc *sch, struct nlattr *opt,
  */
 static void ceetm_attach(struct Qdisc *sch)
 {
-	struct net_device *dev = qdisc_dev(sch);
 	struct ceetm_qdisc *priv = qdisc_priv(sch);
+	struct net_device *dev = qdisc_dev(sch);
 	struct Qdisc *qdisc, *old_qdisc;
 	unsigned int i;
 
@@ -1461,8 +1461,8 @@ static int ceetm_cls_change_prio(struct ceetm_class *cl,
 	}
 
 	if (cl->prio.cr != (bool)copt->cr) {
-		err = qman_ceetm_channel_set_cq_cr_eligibility(
-						cl->prio.cq->parent,
+		err = qman_ceetm_channel_set_cq_cr_eligibility
+						(cl->prio.cq->parent,
 						cl->prio.cq->idx,
 						copt->cr);
 		if (err)
@@ -1471,8 +1471,8 @@ static int ceetm_cls_change_prio(struct ceetm_class *cl,
 	}
 
 	if (cl->prio.er != (bool)copt->er) {
-		err = qman_ceetm_channel_set_cq_er_eligibility(
-						cl->prio.cq->parent,
+		err = qman_ceetm_channel_set_cq_er_eligibility
+						(cl->prio.cq->parent,
 						cl->prio.cq->idx,
 						copt->er);
 		if (err)
@@ -1517,15 +1517,15 @@ static int ceetm_cls_change(struct Qdisc *sch, u32 classid, u32 parentid,
 			    struct nlattr **tca, unsigned long *arg,
 			    struct netlink_ext_ack *extack)
 {
-	int err;
-	u64 bps;
-	struct ceetm_qdisc *priv;
 	struct ceetm_class *cl = (struct ceetm_class *)*arg;
+	struct net_device *dev = qdisc_dev(sch);
 	struct nlattr *opt = tca[TCA_OPTIONS];
 	struct nlattr *tb[__TCA_CEETM_MAX];
-	struct tc_ceetm_copt *copt;
 	struct qm_ceetm_channel *channel;
-	struct net_device *dev = qdisc_dev(sch);
+	struct tc_ceetm_copt *copt;
+	struct ceetm_qdisc *priv;
+	int err;
+	u64 bps;
 
 	pr_debug(KBUILD_BASENAME " : %s : classid %X under qdisc %X\n",
 		 __func__, classid, sch->handle);
@@ -1714,8 +1714,8 @@ static int ceetm_cls_dump(struct Qdisc *sch, unsigned long arg,
 			  struct sk_buff *skb, struct tcmsg *tcm)
 {
 	struct ceetm_class *cl = (struct ceetm_class *)arg;
-	struct nlattr *nest;
 	struct tc_ceetm_copt copt;
+	struct nlattr *nest;
 
 	pr_debug(KBUILD_BASENAME " : %s : class %X under qdisc %X\n",
 		 __func__, cl->common.classid, sch->handle);
@@ -1770,8 +1770,8 @@ static int ceetm_cls_dump(struct Qdisc *sch, unsigned long arg,
 
 static int ceetm_cls_delete(struct Qdisc *sch, unsigned long arg)
 {
-	struct ceetm_qdisc *priv = qdisc_priv(sch);
 	struct ceetm_class *cl = (struct ceetm_class *)arg;
+	struct ceetm_qdisc *priv = qdisc_priv(sch);
 
 	pr_debug(KBUILD_BASENAME " : %s : class %X under qdisc %X\n",
 		 __func__, cl->common.classid, sch->handle);
@@ -1818,12 +1818,12 @@ static int ceetm_cls_graft(struct Qdisc *sch, unsigned long arg,
 static int ceetm_cls_dump_stats(struct Qdisc *sch, unsigned long arg,
 				struct gnet_dump *d)
 {
-	unsigned int i;
 	struct ceetm_class *cl = (struct ceetm_class *)arg;
 	struct gnet_stats_basic_packed tmp_bstats;
 	struct ceetm_class_stats *cstats = NULL;
 	struct qm_ceetm_cq *cq = NULL;
 	struct tc_ceetm_xstats xstats;
+	unsigned int i;
 
 	memset(&xstats, 0, sizeof(xstats));
 	memset(&tmp_bstats, 0, sizeof(tmp_bstats));
@@ -1872,9 +1872,11 @@ static int ceetm_cls_dump_stats(struct Qdisc *sch, unsigned long arg,
 static struct tcf_block *ceetm_tcf_block(struct Qdisc *sch, unsigned long arg,
 					 struct netlink_ext_ack *extack)
 {
-	struct ceetm_qdisc *priv = qdisc_priv(sch);
 	struct ceetm_class *cl = (struct ceetm_class *)arg;
-	struct tcf_block *block = cl ? cl->block : priv->block;
+	struct ceetm_qdisc *priv = qdisc_priv(sch);
+	struct tcf_block *block;
+
+	block = cl ? cl->block : priv->block;
 
 	pr_debug(KBUILD_BASENAME " : %s : class %X under qdisc %X\n", __func__,
 		 cl ? cl->common.classid : 0, sch->handle);
-- 
2.17.1

